internal/grpc/proto/kong/admin/model/v1/vault.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package kong.admin.model.v1;
+
+import "google/api/field_behavior.proto";
+
+option go_package = "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1;v1";
+
+message Vault {
+  string id = 1;
+  int32 created_at = 2;
+  int32 updated_at = 3;
+  string prefix = 4 [(google.api.field_behavior) = REQUIRED];
+  string name = 5 [(google.api.field_behavior) = REQUIRED];
+  string description = 6;
+  message Config {
+    string prefix = 1;
```

Comment:
This only supports `env` vault, but i believe we should support all those available in the DP today, namely `aws`, `gcp` and `hcv`. Each of these have different required configuration and validation.

  https://docs.konghq.com/gateway/latest/kong-enterprise/secrets-management/backends/

Reply:
@GGabriele https://github.com/Kong/koko/pull/464/files#diff-85f2673ddcf702a04020c0c18ca5ee9eb1988643b86742d47a22c74d71d5fdf4R81-R85 to ensure that configuration remains compatible.

Reply:
I'l update this test to catch this.